inspirehep/dojson/hep/rules/bd9xx.py

```diff
@@ -236,20 +236,22 @@ def references2marc(self, key, value):
     journal_artid = get_value(reference, 'publication_info.artid')
     pubnote = build_pubnote(journal_title, journal_volume, journal_pg_start,
                             journal_pg_end, journal_artid)
+    doi = get_value(reference, 'dois[0]')
+
     return {
         '0': get_recid_from_ref(value.get('record')),
         '1': get_value(reference, 'texkey'),
-        'a': get_value(reference, 'dois[0]'),
+        'a': 'doi:{}'.format(doi) if doi else doi,
```

Comment:
if `schema` is `HDL`, you prepend `hdl:` to the `value`. If schema is `URN`, you send back the `value` (that already contains `urn:`).